src/core/interpreter.cc

```diff
@@ -1031,14 +1041,32 @@ int Interpreter::RedisAPCallCommand(lua_State* lua) {
   return reinterpret_cast<Interpreter*>(*ptr)->RedisGenericCommand(false, true);
 }
 
+InterpreterManager::Stats& InterpreterManager::Stats::operator+=(const Stats& other) {
+  this->used_bytes += other.used_bytes;
+  this->interpreter_cnt += other.interpreter_cnt;
+  this->blocked_usec += other.blocked_usec;
+
+  return *this;
+}
+
+InterpreterManager::Stats& InterpreterManager::tl_stats() {
+  static thread_local Stats stats;
+  return stats;
+}
+
 Interpreter* InterpreterManager::Get() {
   // Grow if none is available and we have unused capacity left.
   if (available_.empty() && storage_.size() < storage_.capacity()) {
     storage_.emplace_back();
     return &storage_.back();
   }
 
+  // We can't remove `before` from stats directly because if inspected while we wait it will return
+  // invalid numbers.
+  uint64_t before = absl::GetCurrentTimeNanos() / 1'000;
```

Comment:
i think measuring time is not needed  (at this point at least).
` waker_.await` returns true if it was preempted, false otherwise.
let's just increase the number of events that we had to preempt due to lack of available interpreters. by monitoring the rate of such events we will know how good our interpreter limits are.
if you want we can also measure the non-preempt (false) events as a reference point

Reply:
Putting on my DevOps hat (I don't really have one but, you know..) I actually think that a time unit, even though not perfect of course, is better than a "simple" counter.
If I was in charge of a running server, seeing any non-zero number would puzzle me. Is that a lot? Is that OK?
Thinking in terms of seconds-blocked I can reason about that. How much that affects latency, for example.
However, I've made the change you required :)
LMK if you do think I should revert though.

Reply:
let's start with this version and change if it won't be suffice :)